File: challenges/BackEndChallenge/backend-challenge-java/src/main/java/com/example/backendchallenge/repositories/IntArrayRepository.java
package com.example.backendchallenge.repositories;

import java.util.ArrayList;
import java.util.List;

/**
 * interface contract for integer array repository
 */
public interface IntArrayRepository {

    /**
     * set integer array
     * @param intArray
     */
    void setIntArray(ArrayList<Integer> intArray);

    /**
     * returns sorted integer array
     * @return
     */
    List<Integer> getIntArraySorted();
}

File: challenges/BackEndChallenge/backend-challenge-cs/Repositories/IntArrayRepository.cs
using System.Collections.Generic;

namespace BackendChallengeWebApi.Repositories
{
    public class IntArrayRepository : IIntArrayRepository
    {
        private List<int> intArray = new List<int>();

        public void SetIntArray(List<int> intArray) 
        {
            this.intArray = intArray;
        }

        public List<int> GetIntArraySorted() 
        {
            var intArray = new List<int>(this.intArray);
            intArray.Sort();
            return intArray;
        }
    }
}